overwatch-derive/src/lib.rs

```diff
@@ -280,21 +263,21 @@ fn generate_start_impl(fields: &Punctuated<Field, Comma>) -> proc_macro2::TokenS
     let cases = fields.iter().map(|field| {
         let field_identifier = field.ident.as_ref().expect("A struct attribute identifier");
         let type_id = utils::extract_type_from(&field.ty);
+        let aggregated_service_id_type_name = get_aggregated_service_id_type_name();
         quote! {
-            <#type_id as ::overwatch::services::ServiceData>::SERVICE_ID => {
+            &<#type_id as ::overwatch::services::ServiceId<#aggregated_service_id_type_name>>::SERVICE_ID => {
                 self.#field_identifier.service_runner::<<#type_id as ::overwatch::services::ServiceData>::StateOperator>().run::<#type_id>()?;
                 ::core::result::Result::Ok(())
             }
         }
     });
 
-    let instrumentation = get_default_instrumentation();
+    let instrumentation = get_default_instrumentation_for_result();
     quote! {
         #instrumentation
-        fn start(&mut self, service_id: ::overwatch::services::ServiceId) -> ::core::result::Result<(), ::overwatch::overwatch::Error> {
+        fn start(&mut self, service_id: &Self::AggregatedServiceId) -> ::core::result::Result<(), ::overwatch::overwatch::Error> {
```

Comment:
Can we make the `AggregatedServiceId` just a `ServiceId` (even in a later PR)?

Reply:
Oh, I see it may conflict with the new trait. Also this is the type itself, and since its internal to overwatch `Services` should be fine. No need to change naming, its clear as it is.

Reply:
I am not sure I understand what your suggestion was? I think I will rename the trait itself to something like `RuntimeServiceId`, and all the generics and associated types to the same. I don't like `AggregatedServiceId` that much as it's not very clear.

Reply:
Ok I left the trait `ServiceId`, since it's implemented by services, and renamed the generics and associated types `RuntimeServiceId`. I like it more now, let me know what you think!